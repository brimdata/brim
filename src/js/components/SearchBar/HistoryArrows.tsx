import React from "react"
import styled from "styled-components"

import Back from "../icons/back-arrow"
import Button from "./Button"
import Forward from "../icons/forward-arrow"
import {useHistory} from "react-router"
import TabHistory from "src/app/router/tab-history"
import {useDispatch} from "react-redux"

const Wrap = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: center;
`

export default function HistoryArrows() {
  const history = useHistory()
  const dispatch = useDispatch()
  // @ts-ignore history definitely has an index
  const canGoBack = history.index > 0
  // @ts-ignore history definitely has an index
  const canGoForward = history.length - 1 != history.index
  const back = () => dispatch(TabHistory.goBack())
  const forward = () => dispatch(TabHistory.goForward())

  return (
    <Wrap>
      <Button disabled={!canGoBack} onClick={back}>
        <Back />
      </Button>
      <Button disabled={!canGoForward} onClick={forward}>
        <Forward />
      </Button>
    </Wrap>
  )
}
