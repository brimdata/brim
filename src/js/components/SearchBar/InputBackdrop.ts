
import styled from "styled-components";

import { Styled } from "../../types/styled";

const Input: Styled<> = styled.div`
  display: block;
  outline: none;
  border: none;
  padding: 0;
  border-radius: 15px;
  height: 28px;
  line-height: 28px;
  font-size: 13px;
  letter-spacing: 0.8px;
  box-shadow: 0 0 0 0.5px var(--lead);
  width: 100%;
  position: relative;
  background: white;
  display: flex;
  align-items: center;
`;

export default Input;