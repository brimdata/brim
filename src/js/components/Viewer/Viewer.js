/* @flow */

import * as React from "react"

import type {RowRenderer, ViewerDimens} from "../../types"
import Chunk from "./Chunk"
import Chunker from "./Chunker"
import * as Doc from "../../lib/Doc"
import Header from "./Header"
import Log from "../../models/Log"
import ScrollHooks from "../../lib/ScrollHooks"
import * as Styler from "./Styler"
import TableColumns from "../../models/TableColumns"
import {reactElementProps} from "../../test/integration"

type Props = {
  chunker: Chunker,
  dimens: ViewerDimens,
  tableColumns: TableColumns,
  renderRow: RowRenderer,
  logs: Log[],
  onLastChunk?: Function,
  renderEnd: () => *
}

type State = {
  scrollLeft: number,
  chunks: number[]
}

export default class Viewer extends React.PureComponent<Props, State> {
  scrollHooks: Function
  view: ?HTMLDivElement

  constructor(props: Props) {
    super(props)
    this.state = {
      scrollLeft: 0,
      chunks: props.chunker.visibleChunks(0)
    }
    this.scrollHooks = ScrollHooks.create(this.onScrollStart, this.onScrollStop)
  }

  componentDidUpdate() {
    if (!this.view) return
    this.updateChunks(this.view.scrollTop)
    if (
      this.props.chunker.lastChunk() ==
      this.state.chunks[this.state.chunks.length - 1]
    ) {
      this.props.onLastChunk && this.props.onLastChunk()
    }
  }

  onScrollStart() {
    Doc.id("tooltip-root").style.display = "none"
  }

  onScrollStop() {
    Doc.id("tooltip-root").style.display = "block"
  }

  onScroll = () => {
    this.scrollHooks()
    const view = this.view
    if (view) {
      this.updateChunks(view.scrollTop)
      this.setState({scrollLeft: view.scrollLeft})
    }
  }

  updateChunks(scrollTop: number) {
    const next = this.props.chunker.visibleChunks(scrollTop)
    const current = this.state.chunks
    if (!Chunker.isEqual(next, current)) {
      this.setState({chunks: next})
    }
  }

  render() {
    const {
      dimens,
      chunker,
      renderRow,
      logs,
      tableColumns,
      renderEnd
    } = this.props
    const {scrollLeft, chunks} = this.state
    return (
      <div className="viewer" style={{width: dimens.viewWidth}}>
        <Header
          columns={tableColumns}
          dimens={dimens}
          scrollLeft={scrollLeft}
          locator="viewer_header"
        />
        <div
          className="view"
          onScroll={this.onScroll}
          style={{width: dimens.viewWidth, height: dimens.viewHeight}}
          ref={(r) => (this.view = r)}
        >
          <div
            className="list"
            style={Styler.list(dimens)}
            {...reactElementProps("viewer_results")}
          >
            {chunks.map((chunk) => (
              <Chunk
                columns={this.props.tableColumns}
                logs={logs}
                rows={chunker.rows(chunk)}
                key={chunk}
                rowRenderer={renderRow}
                dimens={dimens}
              />
            ))}
            {renderEnd()}
          </div>
        </div>
      </div>
    )
  }
}
