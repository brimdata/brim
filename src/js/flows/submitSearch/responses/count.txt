{"type":"TaskStart","task_id":0}


{"type":"SearchRecords","channel_id":0,"records":[{"id":46,"type":[{"name":"count","type":"uint64"}],"values":["8100"]}]}


{"type":"SearchEnd","channel_id":0,"reason":"eof"}


{"type":"SearchStats","start_time":{"sec":1598648691,"ns":566035000},"update_time":{"sec":1598648691,"ns":617932000},"bytes_read":4124856,"bytes_matched":3711103,"records_read":11308,"records_matched":8100}


{"type":"TaskEnd","task_id":0}
