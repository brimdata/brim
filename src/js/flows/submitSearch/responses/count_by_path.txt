{"type":"TaskStart","task_id":0}


{"type":"SearchRecords","channel_id":0,"records":[{"id":41,"type":[{"name":"ts","type":"time"},{"name":"_path","type":"string"},{"name":"count","type":"uint64"}],"values":["1585852140","stats","1"]},{"id":41,"values":["1585852140","dns","15"]},{"id":41,"values":["1585852140","conn","2"]}]}


{"type":"SearchRecords","channel_id":0,"records":[{"id":41,"values":["1585852080","dns","121"]},{"id":41,"values":["1585852080","conn","58"]},{"id":41,"values":["1585852080","x509","4"]}]}


{"type":"SearchRecords","channel_id":0,"records":[{"id":41,"values":["1585852020","dns","80"]},{"id":41,"values":["1585852020","conn","12"]}]}


{"type":"SearchRecords","channel_id":0,"records":[{"id":41,"values":["1585851960","conn","16"]},{"id":41,"values":["1585851960","x509","2"]},{"id":41,"values":["1585851960","dns","81"]}]}


{"type":"SearchRecords","channel_id":0,"records":[{"id":41,"values":["1585851900","dns","48"]},{"id":41,"values":["1585851900","conn","17"]}]}


{"type":"SearchRecords","channel_id":0,"records":[{"id":41,"values":["1585851840","dns","38"]},{"id":41,"values":["1585851840","conn","8"]},{"id":41,"values":["1585851840","stats","1"]}]}


{"type":"SearchRecords","channel_id":0,"records":[{"id":41,"values":["1585851780","dns","42"]},{"id":41,"values":["1585851780","conn","10"]}]}


{"type":"SearchRecords","channel_id":0,"records":[{"id":41,"values":["1585851720","dns","49"]},{"id":41,"values":["1585851720","conn","13"]},{"id":41,"values":["1585851720","x509","1"]}]}


{"type":"SearchRecords","channel_id":0,"records":[{"id":41,"values":["1585851660","conn","12"]},{"id":41,"values":["1585851660","x509","1"]},{"id":41,"values":["1585851660","dns","71"]}]}


{"type":"SearchRecords","channel_id":0,"records":[{"id":41,"values":["1585851600","conn","23"]},{"id":41,"values":["1585851600","dns","59"]}]}


{"type":"SearchRecords","channel_id":0,"records":[{"id":41,"values":["1585851540","dns","34"]},{"id":41,"values":["1585851540","conn","10"]},{"id":41,"values":["1585851540","stats","1"]}]}


{"type":"SearchRecords","channel_id":0,"records":[{"id":41,"values":["1585851480","dns","46"]},{"id":41,"values":["1585851480","conn","10"]}]}


{"type":"SearchRecords","channel_id":0,"records":[{"id":41,"values":["1585851420","conn","14"]},{"id":41,"values":["1585851420","dns","86"]}]}


{"type":"SearchRecords","channel_id":0,"records":[{"id":41,"values":["1585851360","dns","149"]},{"id":41,"values":["1585851360","conn","22"]}]}


{"type":"SearchRecords","channel_id":0,"records":[{"id":41,"values":["1585851300","dns","106"]},{"id":41,"values":["1585851300","conn","31"]}]}


{"type":"SearchRecords","channel_id":0,"records":[{"id":41,"values":["1585851240","dns","82"]},{"id":41,"values":["1585851240","conn","22"]},{"id":41,"values":["1585851240","stats","1"]},{"id":41,"values":["1585851240","x509","3"]}]}


{"type":"SearchRecords","channel_id":0,"records":[{"id":41,"values":["1585851180","conn","10"]},{"id":41,"values":["1585851180","dns","71"]}]}


{"type":"SearchRecords","channel_id":0,"records":[{"id":41,"values":["1585851120","conn","15"]},{"id":41,"values":["1585851120","dns","65"]}]}


{"type":"SearchRecords","channel_id":0,"records":[{"id":41,"values":["1585851060","dns","87"]},{"id":41,"values":["1585851060","conn","14"]}]}


{"type":"SearchRecords","channel_id":0,"records":[{"id":41,"values":["1585851000","dns","64"]},{"id":41,"values":["1585851000","conn","15"]}]}


{"type":"SearchRecords","channel_id":0,"records":[{"id":41,"values":["1585850940","dns","39"]},{"id":41,"values":["1585850940","conn","7"]},{"id":41,"values":["1585850940","stats","1"]}]}


{"type":"SearchRecords","channel_id":0,"records":[{"id":41,"values":["1585850880","dns","84"]},{"id":41,"values":["1585850880","conn","11"]}]}


{"type":"SearchRecords","channel_id":0,"records":[{"id":41,"values":["1585850820","conn","10"]},{"id":41,"values":["1585850820","dns","70"]}]}


{"type":"SearchRecords","channel_id":0,"records":[{"id":41,"values":["1585850760","dns","93"]},{"id":41,"values":["1585850760","x509","1"]},{"id":41,"values":["1585850760","conn","15"]}]}


{"type":"SearchRecords","channel_id":0,"records":[{"id":41,"values":["1585850700","dns","42"]},{"id":41,"values":["1585850700","conn","13"]},{"id":41,"values":["1585850700","x509","1"]}]}


{"type":"SearchRecords","channel_id":0,"records":[{"id":41,"values":["1585850640","stats","1"]},{"id":41,"values":["1585850640","dns","74"]},{"id":41,"values":["1585850640","conn","11"]}]}


{"type":"SearchRecords","channel_id":0,"records":[{"id":41,"values":["1585850580","conn","19"]},{"id":41,"values":["1585850580","x509","3"]},{"id":41,"values":["1585850580","dns","87"]}]}


{"type":"SearchRecords","channel_id":0,"records":[{"id":41,"values":["1585850520","conn","10"]},{"id":41,"values":["1585850520","dns","44"]}]}


{"type":"SearchRecords","channel_id":0,"records":[{"id":41,"values":["1585850460","dns","78"]},{"id":41,"values":["1585850460","conn","13"]}]}


{"type":"SearchRecords","channel_id":0,"records":[{"id":41,"values":["1585850400","dns","85"]},{"id":41,"values":["1585850400","conn","14"]}]}


{"type":"SearchRecords","channel_id":0,"records":[{"id":41,"values":["1585850340","dns","79"]},{"id":41,"values":["1585850340","conn","9"]},{"id":41,"values":["1585850340","stats","1"]}]}


{"type":"SearchRecords","channel_id":0,"records":[{"id":41,"values":["1585850280","x509","2"]},{"id":41,"values":["1585850280","dns","86"]},{"id":41,"values":["1585850280","conn","17"]}]}


{"type":"SearchRecords","channel_id":0,"records":[{"id":41,"values":["1585850220","conn","12"]},{"id":41,"values":["1585850220","dns","54"]}]}


{"type":"SearchRecords","channel_id":0,"records":[{"id":41,"values":["1585850160","dns","176"]},{"id":41,"values":["1585850160","conn","34"]},{"id":41,"values":["1585850160","x509","1"]}]}


{"type":"SearchRecords","channel_id":0,"records":[{"id":41,"values":["1585850100","dns","69"]},{"id":41,"values":["1585850100","conn","27"]}]}


{"type":"SearchRecords","channel_id":0,"records":[{"id":41,"values":["1585850040","dns","88"]},{"id":41,"values":["1585850040","conn","49"]},{"id":41,"values":["1585850040","x509","8"]},{"id":41,"values":["1585850040","stats","1"]}]}


{"type":"SearchRecords","channel_id":0,"records":[{"id":41,"values":["1585849980","conn","5"]},{"id":41,"values":["1585849980","dns","35"]}]}


{"type":"SearchRecords","channel_id":0,"records":[{"id":41,"values":["1585849920","dns","26"]},{"id":41,"values":["1585849920","conn","13"]}]}


{"type":"SearchRecords","channel_id":0,"records":[{"id":41,"values":["1585849860","dns","53"]},{"id":41,"values":["1585849860","conn","13"]}]}


{"type":"SearchRecords","channel_id":0,"records":[{"id":41,"values":["1585849800","dns","41"]},{"id":41,"values":["1585849800","conn","8"]}]}


{"type":"SearchRecords","channel_id":0,"records":[{"id":41,"values":["1585849740","conn","5"]},{"id":41,"values":["1585849740","stats","1"]},{"id":41,"values":["1585849740","dns","35"]}]}


{"type":"SearchRecords","channel_id":0,"records":[{"id":41,"values":["1585849680","dns","39"]},{"id":41,"values":["1585849680","conn","8"]}]}


{"type":"SearchRecords","channel_id":0,"records":[{"id":41,"values":["1585849620","dns","31"]},{"id":41,"values":["1585849620","conn","7"]}]}


{"type":"SearchRecords","channel_id":0,"records":[{"id":41,"values":["1585849560","dns","42"]},{"id":41,"values":["1585849560","conn","13"]}]}


{"type":"SearchRecords","channel_id":0,"records":[{"id":41,"values":["1585849500","conn","8"]},{"id":41,"values":["1585849500","dns","40"]}]}


{"type":"SearchRecords","channel_id":0,"records":[{"id":41,"values":["1585849440","conn","10"]},{"id":41,"values":["1585849440","stats","1"]},{"id":41,"values":["1585849440","dns","42"]}]}


{"type":"SearchRecords","channel_id":0,"records":[{"id":41,"values":["1585849380","conn","11"]},{"id":41,"values":["1585849380","dns","44"]},{"id":41,"values":["1585849380","x509","1"]}]}


{"type":"SearchRecords","channel_id":0,"records":[{"id":41,"values":["1585849320","conn","10"]},{"id":41,"values":["1585849320","dns","51"]}]}


{"type":"SearchRecords","channel_id":0,"records":[{"id":41,"values":["1585849260","conn","8"]},{"id":41,"values":["1585849260","dns","36"]}]}


{"type":"SearchRecords","channel_id":0,"records":[{"id":41,"values":["1585849200","dns","37"]},{"id":41,"values":["1585849200","conn","5"]}]}


{"type":"SearchRecords","channel_id":0,"records":[{"id":41,"values":["1585849140","stats","1"]},{"id":41,"values":["1585849140","dns","41"]},{"id":41,"values":["1585849140","conn","10"]}]}


{"type":"SearchRecords","channel_id":0,"records":[{"id":41,"values":["1585849080","conn","9"]},{"id":41,"values":["1585849080","dns","41"]}]}


{"type":"SearchRecords","channel_id":0,"records":[{"id":41,"values":["1585849020","dns","51"]},{"id":41,"values":["1585849020","conn","13"]}]}


{"type":"SearchRecords","channel_id":0,"records":[{"id":41,"values":["1585848960","conn","5"]},{"id":41,"values":["1585848960","dns","33"]}]}


{"type":"SearchRecords","channel_id":0,"records":[{"id":41,"values":["1585848900","conn","15"]},{"id":41,"values":["1585848900","dns","73"]}]}


{"type":"SearchRecords","channel_id":0,"records":[{"id":41,"values":["1585848840","dns","37"]},{"id":41,"values":["1585848840","stats","1"]},{"id":41,"values":["1585848840","conn","6"]}]}


{"type":"SearchRecords","channel_id":0,"records":[{"id":41,"values":["1585848780","dns","48"]},{"id":41,"values":["1585848780","conn","15"]}]}


{"type":"SearchRecords","channel_id":0,"records":[{"id":41,"values":["1585848720","dns","38"]},{"id":41,"values":["1585848720","conn","5"]}]}


{"type":"SearchRecords","channel_id":0,"records":[{"id":41,"values":["1585848660","conn","21"]},{"id":41,"values":["1585848660","dns","104"]}]}


{"type":"SearchRecords","channel_id":0,"records":[{"id":41,"values":["1585848600","conn","10"]},{"id":41,"values":["1585848600","dns","72"]}]}


{"type":"SearchRecords","channel_id":0,"records":[{"id":41,"values":["1585848540","stats","1"]},{"id":41,"values":["1585848540","conn","17"]},{"id":41,"values":["1585848540","dns","110"]}]}


{"type":"SearchRecords","channel_id":0,"records":[{"id":41,"values":["1585848480","dns","74"]},{"id":41,"values":["1585848480","conn","12"]}]}


{"type":"SearchRecords","channel_id":0,"records":[{"id":41,"values":["1585848420","dns","67"]},{"id":41,"values":["1585848420","conn","11"]}]}


{"type":"SearchRecords","channel_id":0,"records":[{"id":41,"values":["1585848360","dns","101"]},{"id":41,"values":["1585848360","conn","10"]}]}


{"type":"SearchRecords","channel_id":0,"records":[{"id":41,"values":["1585848300","dns","86"]},{"id":41,"values":["1585848300","conn","16"]}]}


{"type":"SearchRecords","channel_id":0,"records":[{"id":41,"values":["1585848240","conn","14"]},{"id":41,"values":["1585848240","dns","79"]},{"id":41,"values":["1585848240","stats","1"]}]}


{"type":"SearchRecords","channel_id":0,"records":[{"id":41,"values":["1585848180","x509","3"]},{"id":41,"values":["1585848180","dns","86"]},{"id":41,"values":["1585848180","conn","21"]}]}


{"type":"SearchRecords","channel_id":0,"records":[{"id":41,"values":["1585848120","x509","1"]},{"id":41,"values":["1585848120","dns","94"]},{"id":41,"values":["1585848120","conn","22"]}]}


{"type":"SearchRecords","channel_id":0,"records":[{"id":41,"values":["1585848060","x509","1"]},{"id":41,"values":["1585848060","dns","107"]},{"id":41,"values":["1585848060","conn","17"]}]}


{"type":"SearchRecords","channel_id":0,"records":[{"id":41,"values":["1585848000","dns","83"]},{"id":41,"values":["1585848000","conn","16"]}]}


{"type":"SearchRecords","channel_id":0,"records":[{"id":41,"values":["1585847940","dns","45"]},{"id":41,"values":["1585847940","conn","5"]},{"id":41,"values":["1585847940","stats","1"]}]}


{"type":"SearchRecords","channel_id":0,"records":[{"id":41,"values":["1585847880","dns","136"]},{"id":41,"values":["1585847880","conn","13"]}]}


{"type":"SearchRecords","channel_id":0,"records":[{"id":41,"values":["1585847820","dns","47"]},{"id":41,"values":["1585847820","conn","11"]}]}


{"type":"SearchRecords","channel_id":0,"records":[{"id":41,"values":["1585847760","dns","136"]},{"id":41,"values":["1585847760","conn","19"]}]}


{"type":"SearchRecords","channel_id":0,"records":[{"id":41,"values":["1585847700","dns","66"]},{"id":41,"values":["1585847700","conn","42"]},{"id":41,"values":["1585847700","x509","16"]},{"id":41,"values":["1585847700","files","1"]},{"id":41,"values":["1585847700","ssl","1"]},{"id":41,"values":["1585847640","dns","14"]},{"id":41,"values":["1585847640","conn","4"]},{"id":41,"values":["1585847640","stats","1"]}]}


{"type":"SearchRecords","channel_id":0,"records":[{"id":41,"values":["1585847580","x509","2"]},{"id":41,"values":["1585847580","dns","50"]},{"id":41,"values":["1585847580","conn","15"]}]}


{"type":"SearchRecords","channel_id":0,"records":[{"id":41,"values":["1585847520","conn","6"]},{"id":41,"values":["1585847520","dns","26"]}]}


{"type":"SearchRecords","channel_id":0,"records":[{"id":41,"values":["1585847460","dns","109"]},{"id":41,"values":["1585847460","conn","16"]}]}


{"type":"SearchRecords","channel_id":0,"records":[{"id":41,"values":["1585847400","conn","21"]},{"id":41,"values":["1585847400","dns","83"]}]}


{"type":"SearchRecords","channel_id":0,"records":[{"id":41,"values":["1585847340","conn","3"]},{"id":41,"values":["1585847340","dns","71"]},{"id":41,"values":["1585847340","stats","1"]}]}


{"type":"SearchRecords","channel_id":0,"records":[{"id":41,"values":["1585847280","dns","110"]},{"id":41,"values":["1585847280","conn","22"]},{"id":41,"values":["1585847220","conn","3"]},{"id":41,"values":["1585847220","dns","7"]}]}


{"type":"SearchRecords","channel_id":0,"records":[{"id":41,"values":["1585847160","conn","18"]},{"id":41,"values":["1585847160","dns","54"]}]}


{"type":"SearchRecords","channel_id":0,"records":[{"id":41,"values":["1585847100","dns","54"]},{"id":41,"values":["1585847100","x509","8"]},{"id":41,"values":["1585847100","conn","17"]}]}


{"type":"SearchRecords","channel_id":0,"records":[{"id":41,"values":["1585847040","x509","1"]},{"id":41,"values":["1585847040","stats","1"]},{"id":41,"values":["1585847040","dns","42"]},{"id":41,"values":["1585847040","conn","13"]}]}


{"type":"SearchRecords","channel_id":0,"records":[{"id":41,"values":["1585846980","conn","11"]},{"id":41,"values":["1585846980","dns","44"]}]}


{"type":"SearchRecords","channel_id":0,"records":[{"id":41,"values":["1585846920","x509","1"]},{"id":41,"values":["1585846920","dns","38"]},{"id":41,"values":["1585846920","conn","6"]}]}


{"type":"SearchRecords","channel_id":0,"records":[{"id":41,"values":["1585846860","dns","46"]},{"id":41,"values":["1585846860","conn","16"]},{"id":41,"values":["1585846860","x509","3"]}]}


{"type":"SearchRecords","channel_id":0,"records":[{"id":41,"values":["1585846800","dns","37"]},{"id":41,"values":["1585846800","conn","7"]}]}


{"type":"SearchRecords","channel_id":0,"records":[{"id":41,"values":["1585846740","dns","52"]},{"id":41,"values":["1585846740","conn","15"]},{"id":41,"values":["1585846740","stats","1"]},{"id":41,"values":["1585846680","dns","7"]},{"id":41,"values":["1585846680","conn","7"]}]}


{"type":"SearchRecords","channel_id":0,"records":[{"id":41,"values":["1585846620","dns","86"]},{"id":41,"values":["1585846620","conn","18"]}]}


{"type":"SearchRecords","channel_id":0,"records":[{"id":41,"values":["1585846560","dns","42"]},{"id":41,"values":["1585846560","conn","12"]},{"id":41,"values":["1585846560","x509","5"]}]}


{"type":"SearchRecords","channel_id":0,"records":[{"id":41,"values":["1585846500","dns","114"]},{"id":41,"values":["1585846500","conn","63"]},{"id":41,"values":["1585846500","x509","3"]}]}


{"type":"SearchRecords","channel_id":0,"records":[{"id":41,"values":["1585846440","dns","53"]},{"id":41,"values":["1585846440","conn","7"]},{"id":41,"values":["1585846440","stats","1"]}]}


{"type":"SearchRecords","channel_id":0,"records":[{"id":41,"values":["1585846380","conn","12"]},{"id":41,"values":["1585846380","dns","96"]}]}


{"type":"SearchRecords","channel_id":0,"records":[{"id":41,"values":["1585846320","dns","48"]},{"id":41,"values":["1585846320","conn","15"]}]}


{"type":"SearchRecords","channel_id":0,"records":[{"id":41,"values":["1585846260","conn","61"]},{"id":41,"values":["1585846260","dns","99"]},{"id":41,"values":["1585846260","x509","1"]}]}


{"type":"SearchRecords","channel_id":0,"records":[{"id":41,"values":["1585846200","dns","47"]},{"id":41,"values":["1585846200","conn","15"]}]}


{"type":"SearchRecords","channel_id":0,"records":[{"id":41,"values":["1585846140","conn","4"]},{"id":41,"values":["1585846140","dns","35"]},{"id":41,"values":["1585846140","stats","1"]}]}


{"type":"SearchEnd","channel_id":0,"reason":"eof"}


{"type":"SearchStats","start_time":{"sec":1598648883,"ns":523336000},"update_time":{"sec":1598648883,"ns":577611000},"bytes_read":4124856,"bytes_matched":3711103,"records_read":11308,"records_matched":8100}


{"type":"TaskEnd","task_id":0}
