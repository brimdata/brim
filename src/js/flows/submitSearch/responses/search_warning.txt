{"type":"TaskStart","task_id":0}


{"type":"SearchWarning","warning":"Cut field boo not present in input"}


{"type":"SearchEnd","channel_id":0,"reason":"eof"}


{"type":"SearchEnd","channel_id":0,"reason":"eof"}


{"type":"SearchStats","start_time":{"sec":1598649951,"ns":558286000},"update_time":{"sec":1598649951,"ns":581530000},"bytes_read":4124856,"bytes_matched":4124856,"records_read":11308,"records_matched":11308}


{"type":"TaskEnd","task_id":0}
