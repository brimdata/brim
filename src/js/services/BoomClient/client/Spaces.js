/* @flow */

import type {CreateSpaceBody} from "../types"
import SubClient from "./SubClient"

export default class Spaces extends SubClient {
  list() {
    return this.base.send({
      method: "GET",
      path: "/space"
    })
  }

  get(name: string) {
    return this.base.send({
      method: "GET",
      path: `/space/${encodeURIComponent(name)}`
    })
  }

  create(space: CreateSpaceBody) {
    return this.base.send({
      method: "POST",
      path: "/space",
      payload: space
    })
  }

  delete(name: string) {
    return this.base.send({
      method: "DELETE",
      path: `/space/${encodeURIComponent(name)}`
    })
  }
}
