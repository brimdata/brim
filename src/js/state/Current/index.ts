import reducer from "./reducer"
import * as selectors from "./selectors"

export default {
  ...selectors,
  reducer
}
