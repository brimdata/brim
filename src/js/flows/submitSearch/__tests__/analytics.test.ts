import {createZealotMock, zng} from "zealot"

import {submitSearch} from "../mod"
import Current from "../../../state/Current"
import Handlers from "../../../state/Handlers"
import SearchBar from "../../../state/SearchBar"
import Spaces from "../../../state/Spaces"
import Viewer from "../../../state/Viewer"
import fixtures from "../../../test/fixtures"
import initTestStore from "../../../test/initTestStore"
import responses from "../../../test/responses"
import Workspaces from "../../../state/Workspaces"

const countResp = responses("count.txt")
const space = fixtures("space1")

let store, zealot, dispatch, select
beforeEach(() => {
  zealot = createZealotMock()
  store = initTestStore(zealot.zealot)
  dispatch = store.dispatch
  select = (s: any) => s(store.getState())
  store.dispatchAll([
    Workspaces.add({
      host: "testHost",
      id: "1",
      name: "testName",
      port: "9867",
      authType: "none"
    }),
    Current.setWorkspaceId("1"),
    Spaces.setDetail("1", space),
    Current.setSpaceId(space.id),
    SearchBar.changeSearchBarInput("dns"),
    SearchBar.pinSearchBar(),
    SearchBar.changeSearchBarInput("query")
  ])
})
const submit = (...args) => dispatch(submitSearch(...args))

describe("analytic search", () => {
  beforeEach(() => {
    dispatch(SearchBar.changeSearchBarInput("_path=conn | count()"))
    zealot.stubStream("search", countResp)
  })

  test("zealot gets the right args", async () => {
    await submit()

    expect(zealot.calls("search").length).toBe(1)
    expect(zealot.calls("search")[0].args).toEqual(
      "dns _path=conn | count() | head 10000"
    )
  })

  test("the table gets populated", async () => {
    await submit()
    expect(select(Viewer.getViewerRecords)).toMatchSnapshot()
  })

  test("the table gets cleared", async () => {
    dispatch(
      Viewer.setRecords(undefined, [
        new zng.Record([{name: "clear", type: "string"}], ["me"])
      ])
    )
    submit()
    expect(select(Viewer.getViewerRecords)).toEqual([])
  })

  test("the table status updates", async () => {
    const promise = submit()
    expect(select(Viewer.getStatus)).toBe("FETCHING")
    expect(select(Viewer.getEndStatus)).toBe("FETCHING")
    await promise
    expect(select(Viewer.getStatus)).toBe("SUCCESS")
    expect(select(Viewer.getEndStatus)).toBe("COMPLETE")
  })

  test("registers a table request then cleans it up", async () => {
    const promise = submit()
    expect(select(Handlers.get)["Table"]).toEqual(
      expect.objectContaining({type: "SEARCH"})
    )
    await promise
    expect(select(Handlers.get)["Table"]).toBe(undefined)
  })

  test("aborts previous table request", async () => {
    const abort = jest.fn()
    dispatch(Handlers.register("Table", {type: "SEARCH", abort}))
    await submit()
    expect(abort).toHaveBeenCalledTimes(1)
  })
})
