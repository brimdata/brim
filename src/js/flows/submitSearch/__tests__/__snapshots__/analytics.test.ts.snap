// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`analytic search the table gets populated 1`] = `
Array [
  Record {
    "type": Array [
      Object {
        "name": "count",
        "type": "count",
      },
    ],
    "value": Array [
      "30",
    ],
  },
]
`;
